sessionctx/variable/session.go

```diff
@@ -484,6 +490,12 @@ type SessionVars struct {
 	MetricSchemaStep int64
 	// MetricSchemaRangeDuration indicates the step when query metric schema.
 	MetricSchemaRangeDuration int64
+
+	// Some data of cluster-level memory tables will be retrieved many times in different inspection rules,
+	// and the cost of retrieving some data is expensive. We use the `TableSnapshot` to cache those data
+	// and obtain them lazily, and provide a consistent view of inspection tables for each inspection rules.
+	// All cached snapshots will be released at the `InspectionExec` executor closing.
+	InspectionTableCache map[string]TableSnapshot
```

Comment:
Do you mean that the snapshot is only cached in each session, and different sessions don't share the cache? Isn't it inefficient?

Reply:
@djshow832 Yes, each session may `SELECT * FROM inspection_result` at a different time and the result may be different. And the `InspectionTableCache` is a very short lives data.